tox.ini

```
@@ -0,0 +1,73 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+[tox]
+skipsdist=True
+skip_missing_interpreters = True
+envlist = install, lint, snap, unit
+
+[vars]
+snap_path = {toxinidir}/snap/
+tst_path = {toxinidir}/tests/
+all_path = {[vars]tst_path}
+
+[testenv]
+setenv =
+    PYTHONPATH = {toxinidir}:{[vars]tst_path}
+    PYTHONBREAKPOINT=ipdb.set_trace
+    PY_COLORS=1
+passenv =
+    PYTHONPATH
+allowlist_externals =
+    /snap/bin/snapcraft
+    /usr/bin/snap
+    bash
+    rm
+    sudo
+
+[testenv:fmt]
+description = Apply coding style standards to code
+deps =
+    black
+    ruff
+commands =
+    black {[vars]all_path}
+    ruff --fix {[vars]all_path}
+
+[testenv:lint]
+description = Check code against coding style standards
+deps =
+    black
+    codespell
+    ruff
+    yamllint
+commands =
+    black --check --diff {[vars]all_path}
+    codespell {toxinidir}
+    ruff {[vars]all_path}
+    yamllint {[vars]snap_path}
+
+[testenv:snap]
+description = Build GLAuth snap package using LXD as build provider.
+commands =
+    snapcraft
```

Generally I just like to specify verbose output so I can scroll back through the log if any errors are thrown during the package build:

```suggestion
    snapcraft -v
```